File: app/src/main/java/com/tohami/egyptinnovate/dagger/module/ActivityBindingModule.java
package com.tohami.egyptinnovate.dagger.module;

import com.tohami.egyptinnovate.ui.main.MainActivity;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ActivityBindingModule {

    @ContributesAndroidInjector(modules = {MainActivityBindingModule.class})
    abstract MainActivity bindMainActivity();
}

File: app/src/main/java/com/tohami/egyptinnovate/dagger/module/ActivityBindingModule.kt
package com.tohami.egyptinnovate.dagger.module

import com.tohami.egyptinnovate.ui.main.MainActivity

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ActivityBindingModule {

    @ContributesAndroidInjector(modules = [MainActivityBindingModule::class])
    internal abstract fun bindMainActivity(): MainActivity
}